src/cmd/run.js

```diff
@@ -272,6 +293,21 @@ export default async function run(
         );
       }
 
+      if (stdin.isTTY && stdin instanceof tty.ReadStream) {
+        readline.emitKeypressEvents(stdin);
+        log.info('Press R to reload (and Ctrl-C to quit)');
+        stdin.setRawMode(true);
+        stdin.on('keypress', (str, key) => {
```

Comment:
I've been thinking a bit about this, and I'm wondering if this can be rewritten into an `async function` and a `while` loop (mostly like we are planning to do in the 'web-ext create' PR), e.g. something like:

```js
if (stdin.isTTY && stdin instanceof tty.ReadStream) {
  readline.emitKeypressEvents(stdin);
  stdin.setRawMode(true);

  // NOTE: this `Promise.resolve().then(...)` is basically used to spawn a "co-routine" that is executed
  // before the callback attached to the Promise returned by this function (and it allows the `run` function
  // to not be stuck in the while loop).
  Promise.resolve().then(async function() {
    log.info('Press R to reload (and Ctrl-C to quit)');

    let userExit = false;

    while (!userExit) {
      const keyPressed = await new Promise((resolve) => {
        stdin.once('keypress', (str, key) => resolve(key));
      });

      if (keyPressed.ctrl && keyPressed.name === 'c') {
        userExit = true;
      } else if (keyPressed.name === 'r' && addonId) {
        await addonReload({addonId, client});
      }
    }

    log.info('\nExiting web-ext on user request');
    runningFirefox.kill();
    stdin.pause();
  });
}
```

The above version has also the following advantages:
- no listener to the stdin can be leaked
- if the user press 'r' multiple times in fast sequence, it will not generate a queue of addon reloads request (because the keypress listener is not subscribed until the next loop, after the addon has been restarted).